### ISSUE TITLE
Feature Suggestion: CD

### ISSUE THREAD

**Issue by Javyre** [gittup/tup#356]
I understand the convenience of having a Tupfile per directory but don't want to polute my entire directory tree with Tupfiles as it can become hard to keep track of.

What about a `cd dir` command in Tupfiles that changes the current working directory of the definitions from that point on and acts as if the rest of the contents of the Tupfile are in a seprate Tupfile inside `dir`

Before: 

`/Tupfile`:
```tup
CFLAGS += -O2
src/*.o |> clang $(CFLAGS) -o %o %f |> program
```
`/src/Tupfile`:
```tup
foreach *.c |> clang -c %f |> %B.o
```

After:

`/Tupfile`:
```tup
CFLAGS += -O2
src/*.o |> clang $(CFLAGS) -o %o %f |> program

cd src
foreach *.c |> clang $(CFLAGS) -c %f |> %B.o
```

I have no idea how radical of a change this would be in tups code but I would be ready to try to implement this myself if you could point me in the right direction

**Comment by tehKaiN** [comment]
Can't you just write something like this?

```
CFLAGS += -O2
foreach src/*.c |> clang -c %f |> src/%B.o
src/*.o |> clang $(CFLAGS) -o %o %f |> program
```

Note the `src/` in input and output of first rule. I'm heavily using such construct. Or am I missing your point?

**Comment by Javyre** [comment]
Yes I do that too, but this quickly turns into a mess and is very hard to maintain

The point is to have this feature while maintaining tup's (imo very very good) readability
